index.js

```diff
@@ -1,102 +1,96 @@
 
 	if (fullname) {
-		cb(null, fullname);
-		return;
+		return Promise.resolve(fullname);
 	}
 
-	npmconf.load(function (err, conf) {
+	return pify(npmconf.load, Promise)().then(function (conf) {
 		fullname = conf.get('init.author.name');
 
-		if (err || !fullname) {
-			fallback(cb);
-			return;
+		if (!fullname) {
+			return fallback();
 		}
 
-		cb(null, fullname);
-	});
+		return fullname;
+	}).catch(fallback);
 };
 
-function fallback (cb) {
+function fallback() {
 	if (process.platform === 'darwin') {
-		exec('id -P', function (err, stdout) {
-			fullname = stdout.trim().split(':')[7];
+		return exec('id -P')
+			.then(function (stdout) {
+				fullname = stdout.trim().split(':')[7];
 
-			// `id -P` should never fail as far as I know, but just in case:
-			if (err || !fullname) {
-				exec('osascript -e "long user name of (system info)"', function (err, stdout) {
-					if (err) {
-						cb();
-						return;
-					}
+				if (!fullname) {
+					throw new Error('Name could not be found.');
+				}
 
-					fullname = stdout.trim();
+				return fullname;
+			})
+			.catch(function () {
+				// `id -P` should never fail as far as I know, but just in case:
+				return exec('osascript -e "long user name of (system info)"')
+					.then(function (stdout) {
+						fullname = stdout.trim();
 
-					cb(null, fullname);
-				});
-				return;
-			}
-
-			cb(null, fullname);
-		});
-
-		return;
+						return fullname;
+					});
+			});
 	}
 
 	if (process.platform === 'win32') {
 		// try git first since fullname is usually not set by default in the system on Windows 7+
-		exec('git config --global user.name', function (err, stdout) {
-			fullname = stdout.trim();
-
-			if (err || !fullname) {
-				exec('wmic useraccount where name="%username%" get fullname', function (err, stdout) {
-					if (err) {
-						cb();
-						return;
-					}
-
-					fullname = stdout.replace('FullName', '').trim();
+		return exec('git config --global user.name')
+			.then(function (stdout) {
+				fullname = stdout.trim();
 
-					cb(null, fullname);
-				});
-				return;
-			}
+				if (!fullname) {
+					throw new Error('Name could not be found.');
+				}
 
-			cb(null, fullname);
-		});
+				return fullname;
+			})
+			.catch(function () {
+				return exec('wmic useraccount where name="%username%" get fullname')
+					.then(function (stdout) {
+						fullname = stdout.replace('FullName', '').trim();
 
-		return;
+						return fullname;
+					});
+			});
 	}
 
-	exec('getent passwd $(whoami)', function (err, stdout) {
-		fullname = (stdout.trim().split(':')[4] || '').replace(/,.*/, '');
+	return exec('getent passwd $(whoami)')
+		.then(function (stdout) {
+			fullname = (stdout.trim().split(':')[4] || '').replace(/,.*/, '');
 
-		if (err || !fullname) {
-			exec('git config --global user.name', function (err, stdout) {
-				if (err) {
-					cb();
-					return;
-				}
-
-				fullname = stdout.trim();
+			if (!fullname) {
+				throw new Error('Name could not be found.');
```

Comment:
Why throw here, though, when you just catch it right away? Could just have the logic here instead of in the catch.